libcxx/test/std/ranges/range.adaptors/range.enumerate/types.h

```diff
@@ -0,0 +1,135 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef TEST_STD_RANGES_RANGE_ADAPTORS_RANGE_ENUMERATE_TYPES_H
+#define TEST_STD_RANGES_RANGE_ADAPTORS_RANGE_ENUMERATE_TYPES_H
+
+#include <cstddef>
+#include <ranges>
+#include <tuple>
+
+#include "test_iterators.h"
+#include "test_macros.h"
+
+// Types
```

Comment:
Are there not already archetypes like these in `test/support`? 
I would really rather not duplicate them if-so.

Additionally, I don't love the relative include of such a header with such a non-descript name.
If these types aren't reusable, could you explain why? And if they are, could we put them in `test/support` and give the header a better name.